File: Java/dijkstra/llama_3_1_8b.java

import java.io.BufferedReader;
import java.io.FileReader;
import java.io.IOException;
import java.util.ArrayList;
import java.util.Arrays;
import java.util.List;
import java.util.PriorityQueue;

class PriorityQueueNode implements Comparable<PriorityQueueNode> {
    int vertex;
    int distance;

    public PriorityQueueNode(int vertex, int distance) {
        this.vertex = vertex;
        this.distance = distance;
    }

    @Override
    public int compareTo(PriorityQueueNode other) {
        return Integer.compare(this.distance, other.distance);
    }
}

public class llama_3_1_8b {
    private static final int INF = Integer.MAX_VALUE;

    public static void dijkstra(int[][] graph, int startVertex) {
        int numVertices = graph.length;
        int[] distances = new int[numVertices];
        boolean[] visited = new boolean[numVertices];

        Arrays.fill(distances, INF);
        distances[startVertex] = 0;

        PriorityQueue<PriorityQueueNode> priorityQueue = new PriorityQueue<>();
        priorityQueue.add(new PriorityQueueNode(startVertex, 0));

        while (!priorityQueue.isEmpty()) {
            PriorityQueueNode currentNode = priorityQueue.poll();
            int vertex = currentNode.vertex;
            int distance = currentNode.distance;

            if (visited[vertex]) {
                continue;
            }

            visited[vertex] = true;

            for (int neighbor = 0; neighbor < numVertices; neighbor++) {
                if (graph[vertex][neighbor] != 0 && !visited[neighbor]) {
                    int newDistance = distance + graph[vertex][neighbor];
                    if (newDistance < distances[neighbor]) {
                        distances[neighbor] = newDistance;
                        priorityQueue.add(new PriorityQueueNode(neighbor, newDistance));
                    }
                }
            }
        }

        //System.out.println(Arrays.toString(distances));
    }

    public static void main(String[] args) {
        String filePath = "../../graph_input.txt";
        List<int[]> edges = new ArrayList<>();
        int maxNode = 0;

        // Leitura do arquivo
        try (BufferedReader br = new BufferedReader(new FileReader(filePath))) {
            String line;
            while ((line = br.readLine()) != null) {
                String[] tokens = line.trim().split("\\s+");
                if (tokens.length != 3) continue;

                int from = Integer.parseInt(tokens[0]);
                int to = Integer.parseInt(tokens[1]);
                int weight = Integer.parseInt(tokens[2]);

                edges.add(new int[]{from, to, weight});
                maxNode = Math.max(maxNode, Math.max(from, to));
            }
        } catch (IOException e) {
            return;
        }

        int numNodes = maxNode + 1;
        int[][] graph = new int[numNodes][numNodes];

        for (int[] edge : edges) {
            int from = edge[0];
            int to = edge[1];
            int weight = edge[2];
            graph[from][to] = weight;
        }

        int source = 0; // Altere se necessário
        dijkstra(graph, source);
    }

}

File: Python/dijkstra/llama-3-1-8b.py

import heapq

class Graph:
    def __init__(self, num_vertices):
        self.num_vertices = num_vertices
        self.adj_list = [[] for _ in range(num_vertices)]

    def add_edge(self, u, v, weight):
        self.adj_list[u].append((v, weight))

    def dijkstra(self, start_vertex):
        distances = [float('inf')] * self.num_vertices
        distances[start_vertex] = 0
        pq = [(0, start_vertex)]

        while pq:
            current_distance, current_vertex = heapq.heappop(pq)

            if current_distance > distances[current_vertex]:
                continue

            for neighbor, weight in self.adj_list[current_vertex]:
                distance = current_distance + weight

                if distance < distances[neighbor]:
                    distances[neighbor] = distance
                    heapq.heappush(pq, (distance, neighbor))

        return distances


def main():
    import sys
    input_path = "graph_input.txt" if len(sys.argv) < 2 else sys.argv[1]

    try:
        with open(input_path, 'r') as f:
            edges = [line.strip().split() for line in f if line.strip()]
    except FileNotFoundError:
        return
    edges = [(int(u), int(v), int(w)) for u, v, w in edges]

    all_nodes = {u for u, v, _ in edges} | {v for u, v, _ in edges}
    num_vertices = max(all_nodes) + 1

    g = Graph(num_vertices)

    for u, v, w in edges:
        g.add_edge(u, v, w)
        g.add_edge(v, u, w)  

    start_node = edges[0][0]
    distances = g.dijkstra(start_node)

if __name__ == "__main__":
    main()